net/davix/src/TDavixFile.cxx

```diff
@@ -142,6 +150,118 @@ static void ConfigureDavixLogLevel()
    }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+
+bool normalizeToken(const std::string &input_token, std::string &output_token)
+{
+	static const std::string whitespace = " \t\f\n\v\r";
+		// Tokens are meant for use in an HTTP header; these are forbidden
+		// characters.
+	static const std::string nonheader_whitespace = "\r\n";
+	auto begin = input_token.find_first_not_of(whitespace);
+		// No token here, but not an error.
+	if (begin == std::string::npos) {
+		output_token = "";
+		return true;
+	}
+
+	std::string token = input_token.substr(begin);
+	auto end = token.find_last_not_of(whitespace);
+	token = token.substr(0, end + 1);
+
+	// If non-permitted header characters are present ("\r\n"),
+	// then this token is not permitted.
+	if (token.find(nonheader_whitespace) != std::string::npos) {
+		output_token = "";
+    R__LOG_ERROR(TDavixLogChannel()) << "Token discovery failure: token contains non-permitted character sequence (\\r\\n)";
+		return false;
+	}
+
+	output_token = token;
+	return true;
+}
+
+bool findTokenInFile(const std::string &token_file, std::string &output_token)
+{
+  R__LOG_INFO(TDavixLogChannel()) << "Looking for token in file " << token_file.c_str();
+  auto fd = fopen(token_file.c_str(), "rb");
+  if (fd == nullptr) {
+     output_token = "";
+     if (errno == ENOENT) {
+       return true;
+     }
+     R__LOG_ERROR(TDavixLogChannel()) << "Cannot open '" << token_file << "', error: " << strerror(errno);
+     return false;
+   }
+
+	// As a pragmatic matter, we limit tokens to 16KB; this is often the limit
+	// in size of HTTP headers for many web servers.
+	// This also avoids unbounded memory use in case if the user points us to
+	// a large file.
+	static const size_t max_size = 16384;
+	std::vector<char> input_buffer;
+	input_buffer.resize(max_size);
+  unsigned char *buf = (unsigned char*)malloc(ftell(fd));
+	ssize_t retval = fread(buf, sizeof(char), max_size, fd);
+	fclose(fd);
+  free(buf);
+
+	if (retval == -1) {
+		output_token = "";
+    R__LOG_ERROR(TDavixLogChannel()) << "Token discovery failure: failed to read file " << token_file.c_str() << "', error: " << strerror(errno);
+		return false;
+	}
+	if (retval == 16384) {
+		// Token may have been truncated!  Erring on the side of caution.
+    R__LOG_ERROR(TDavixLogChannel()) << "Token discovery failure: token was larger than 16KB limit.";
+		return false;
+	}
+
+	std::string token(&input_buffer[0], retval);
+
+	return normalizeToken(token, output_token);
+}
+
+// Token discovery process is based on WLCG specification:
+// https://github.com/WLCG-AuthZ-WG/bearer-token-discovery/blob/master/specification.md
+std::string discoverToken()
+{
+	const char *bearer_token = getenv("BEARER_TOKEN");
+	std::string token;
+	if (bearer_token && *bearer_token)
+	{
+		if (!normalizeToken(bearer_token, token)) {return "";}
```

Comment:
This file might be contains tab character, if it is the case, can you replace them with (3) spaces?

Reply:
@pcanal ops and thanks!